ui/base/src/main/java/org/cru/godtools/base/ui/util/ModelUtils.kt

```diff
@@ -19,3 +21,17 @@ fun Translation?.getTagline(tool: Tool?, context: Context?) =
     this?.let { (tagline ?: description)?.applyTypefaceSpan(getTypeface(context)) } ?: tool?.description ?: ""
 
 private fun Translation.getTypeface(context: Context?) = context?.getTypeface(languageCode)
+
+// region Tool Category
+@JvmName("getToolCategory")
+fun Tool?.getCategory(context: Context, locale: Locale? = null) =
+    this?.category?.let { c -> (locale?.let { context.localize(it) } ?: context).getToolCategoryStringRes(c) ?: c }
```

Comment:
Is this saying that `this` can be null?

Reply:
yeah, when I define the extension function I specified it can be called on a null `Tool` object.

Reply:
so something like this would be valid and execute this function:
```kotlin
val tool: Tool? = null
tool.getCategory(context, locale)
```

the nullability here really matters because it is being called from data binding, which can call this method with a `null` Tool object